### DISCUSSION TITLE
Running SPARQL against a program -> goslin sparql

### DISCUSSION THREAD

**Discussion by JervenBolleman** [eclipse-rdf4j/rdf4j#3417] [Show and tell]
During [elixir-europe/biohackathon2021](https://github.com/elixir-europe/biohackathon-projects-2021) @nilshoffmann finished an implementation of a SPARQL endpoint that returns the results of a computation. 

The code live is opensource and on github: https://github.com/lifs-tools/goslin-sparql. 

You can try it out at [the dev site](https://apps.lifs-tools.org/goslin-sparql-dev/sparql/?query=PREFIX%20goslin%3A%20%3Chttps%3A%2F%2Fidentifiers.org%2Flipids%2Fnomenclature%2F%3E%0ASELECT%20%3Fstring%0AWHERE%20{%20[]%20goslin%3Aany%20%27Cer(d18%3A1%2F20%3A2)%27%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20goslin%3AclassName%20%3Fstring%20.%20}).

It has infinite possibilities for inputs so if there is no `goslin:any` (or equivalent prediate) used no results are returned.

**Comment by abrokenjester** [comment]
Sounds interesting - can you tell a little bit more about what this is exactly, and how one would use it?
